clients/java/zts/src/main/java/com/yahoo/athenz/zts/ZTSClient.java

```diff
@@ -1278,6 +1278,18 @@ public Identity postInstanceRefreshRequest(String domain, String service, Instan
             throw new ZTSClientException(ZTSClientException.BAD_REQUEST, ex.getMessage());
         }
     }
+
+
+    /**
+     * For a given domain and role return AWS credential provider
+     *
+     * @param domainName name of the domain
+     * @param roleName   is the name of the role
+     * @return AWSCredentialsProviderImpl AWS credential provider
+     */
+    public AWSCredentialsProviderImpl getAWSCredentialProvider(String domainName, String roleName) {
+        return new AWSCredentialsProviderImpl(this, domainName, roleName);
```

Comment:
We need to improve the documentation for the API to clarify to the user that the ZTS client must not be closed while the credentials provider is being used or an alternative implementation must be considered where we make a copy of the zts client object. Otherwise when the caller closes the zts client object, our provider will not longer function properly.